text_mining.py

```diff
@@ -0,0 +1,357 @@
+club sentiment anaylsis data for each of the EPL clubs.
+"""
+
+#  Data Sources
+import twitter
+import wikipedia
+from bs4 import BeautifulSoup
+import warnings
+
+
+# Text Analyzer
+from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
+
+import doctest
+import datetime
+import re
+from pickle import dump, load
+from os.path import exists
+from os import mkdir
+
+
+def get_credentials(filename):
+    """
+    Returns credentials for twitter api from file with filename
+    """
+    f = open('keys.txt')
+    keys = f.readlines()
+    # codes from Oliver Steele's Piazza Class notes : keeping secrets
+    credentials = []
+    try:
+        for line in keys:
+            credentials.append(line.strip())
+    finally:
+        f.close()
+    return credentials
+
+
+def save_epl_members_list(filename, api):
+    """
+    Retrieves sequence of twitter.user.User instances of
+    "Premier League Players" Twitter list(public list by BBC) as a list
+    (This list might not be completely up to date, so not every member
+    of this list still players in the EPL)
+    Then Pickles the twitter user instance for each player.
+    """
+    epl_players = api.GetListMembers(None, 'premier-league-players',
+                                     None, 'BBCSport')
+    f = open(filename, 'wb')
+    dump(len(epl_players), f)  # saves the number of players
+    try:
+        for each_player in epl_players:
+            dump(each_player, f)
+    finally:
+        f.close()
+
+
+def load_epl_members_list(filename):
+    """
+    Returns a list of previous pickled user instances of EPL players
+    """
+    f = open(filename, 'rb')
+    players = []
+    num_players = load(f)  # loads the number of players
+    try:
+        for i in range(num_players):
+            players.append(load(f))
+    finally:
+        f.close()
+        return players
+
+
+def save_tweets(player_id, player_name, api):
+    """
+    Pickles file "players_status"/player_name.pickle" of
+    200 latest Status Instances of player with player_id.
+    200 is the maximum number of tweets limited by Twitter Api.
+    """
+    #  pickles the latest 200(max limit) statuses for each player
+    statuses = api.GetUserTimeline(user_id=player_id, count=200)
+    if not exists("players_status"):
+        mkdir("players_status")
+    f = open("players_status/" + player_name, 'wb')
+    dump(len(statuses), f)  # saves the number of total Tweets
+    try:
+        for status in statuses:
+            dump(status, f)
+    finally:
+        f.close()
+
+
+def load_tweets(player_name):
+    """
+    Returns a list of all pickled Status Instances of player with player_name
+    """
+    f = open("players_status/" + player_name, 'rb')
+    statuses = []
+    num_status = load(f)  # loads the number of players
+    try:
+        for i in range(num_status):
+            statuses.append(load(f))
+    finally:
+        f.close()
+        return statuses
+
```

Comment:
Good modularization of your functionalities. I like how each function has specific purpose and doesn't try to do too much.